.editorconfig

```diff
@@ -267,6 +267,49 @@ dotnet_diagnostic.SA1026.severity = warning
 dotnet_diagnostic.SA1027.severity = none # RSPEC-105
 dotnet_diagnostic.SA1028.severity = warning
 
+# Readability Rules (SA1100-) https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/1.1.118/DOCUMENTATION.md
+
+dotnet_diagnostic.SA1100.severity = warning
+dotnet_diagnostic.SA1101.severity = none    # Doesn't match our coding style, we don't use "this." when not neded
+dotnet_diagnostic.SA1102.severity = warning
+dotnet_diagnostic.SA1103.severity = warning
+dotnet_diagnostic.SA1104.severity = warning
+dotnet_diagnostic.SA1105.severity = warning
+dotnet_diagnostic.SA1106.severity = warning
+dotnet_diagnostic.SA1107.severity = warning
+dotnet_diagnostic.SA1108.severity = none    # Noisy for short comments on short lines (if, for, foreach)
+dotnet_diagnostic.SA1109.severity = warning
+dotnet_diagnostic.SA1110.severity = warning
+dotnet_diagnostic.SA1111.severity = warning
+dotnet_diagnostic.SA1112.severity = warning
+dotnet_diagnostic.SA1113.severity = warning
+dotnet_diagnostic.SA1114.severity = none    # Prevents putting comment before first member in ImmutableArray.Create
+dotnet_diagnostic.SA1115.severity = warning
+dotnet_diagnostic.SA1116.severity = none    # Waste of new lines in simple scenarios
+dotnet_diagnostic.SA1117.severity = none    # Waste of new lines in simple scenarios
+dotnet_diagnostic.SA1118.severity = none    # Noisy in UTs
+dotnet_diagnostic.SA1119.severity = warning
+dotnet_diagnostic.SA1120.severity = warning
+dotnet_diagnostic.SA1121.severity = warning
+dotnet_diagnostic.SA1122.severity = warning
+dotnet_diagnostic.SA1123.severity = warning
+dotnet_diagnostic.SA1124.severity = none    # We need regions sometimes
+dotnet_diagnostic.SA1125.severity = warning
+dotnet_diagnostic.SA1126.severity = none    # Deprecated / not implemented rule
+dotnet_diagnostic.SA1127.severity = none    # Noisy for single-line method declarations
+dotnet_diagnostic.SA1128.severity = none    # Doesn't match our code base
+dotnet_diagnostic.SA1129.severity = warning
+dotnet_diagnostic.SA1130.severity = warning
+dotnet_diagnostic.SA1131.severity = warning
+dotnet_diagnostic.SA1132.severity = warning
+dotnet_diagnostic.SA1133.severity = warning
+dotnet_diagnostic.SA1134.severity = warning
+dotnet_diagnostic.SA1135.severity = none    # Noisy for generics
```

Comment:
> A violation of this rule occurs when a using directive is contained within a namespace and is not qualified.

but we don't put usings inside namespaces at all, right? maybe update the comment, to reflect this?

Reply:
This rule goes crazy on some special cases that I find useful:

https://github.com/SonarSource/sonar-dotnet/blob/36b38ad5a4f08f580e3ccd90e0b1d960d3d2c701/analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs#L35-L36

https://github.com/SonarSource/sonar-dotnet/blob/36b38ad5a4f08f580e3ccd90e0b1d960d3d2c701/analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs#L38-L39

Reply:
It's not fully qualified, but it's way more readable than the fully qualified version.

Reply:
Ok. This is an interesting pattern, I haven't seen it before I think. I agree